src/mono/mono/metadata/appdomain.c

```diff
@@ -1239,15 +1249,23 @@ void
 mono_runtime_register_appctx_properties (int nprops, const char **keys,  const char **values)
 {
 	n_appctx_props = nprops;
-	appctx_keys = g_new0 (gunichar2*, nprops);
-	appctx_values = g_new0 (gunichar2*, nprops);
+	appctx_keys = g_new0 (char *, n_appctx_props);
+	appctx_values = g_new0 (char *, n_appctx_props);
 
 	for (int i = 0; i < nprops; ++i) {
-		appctx_keys [i] = g_utf8_to_utf16 (keys [i], strlen (keys [i]), NULL, NULL, NULL);
-		appctx_values [i] = g_utf8_to_utf16 (values [i], strlen (values [i]), NULL, NULL, NULL);
+		appctx_keys [i] = g_strdup (keys [i]);
```

Comment:
I suspect long-term we're actually going to want to do a shallow copy here, now that I think about it. @lambdageek thoughts?

This is fine for now though, we can revisit later.

Reply:
@CoffeeFlux no, a shallow copy would be wrong here.  This function is called from `monovm_initialize` with arguments that come from its caller.  We can't just stash them away in a global since we don't know how long the caller will be keeping them alive.  We _must_ do a deep copy.

Reply:
Right. I wonder if we'll want to include a cleanup overload on the `monovm_initialize` replacement then... I think we're going to want to avoid this. Problem for the future though.